react/react_testing/mocking_callbacks_and_components.md

```diff
@@ -0,0 +1,177 @@
+### Introduction
+
+We've already covered some basics of React testing. Now's the time to dive deeper.
+
+In this lesson, we'll learn about mocking. Furthermore, we'll discuss a React component from this ([theodinproject.com](https://theodinproject.com)) website and understand how React tests are written in a real world application.
+
+### Lesson Overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+* Carry out mocks in the context of React testing.
+
+### What is Mocking?
+
+If you've been following along with our lessons so far, the concept of mocking has already been introduced in an earlier section and you might have even incorporated mocks in your [Battleship project in the Testing JavaScript section of this course](https://www.theodinproject.com/lessons/javascript-battleship). Let's look at how mocks will help in testing React components.
+
+#### Testing Callback Handlers
+
+Callbacks are ubiquitous. Every avenue of user interaction involves callbacks. Sometimes they're passed in as props to alter state of the parent component. Consider this simple input component:
+
+~~~jsx
+// FavoriteInput.jsx
+
+import React from "react";
+
+const FavoriteInput = ({ onChange: onInputChange, id }) => {
```

Comment:
I think the prop rename here is unnecessary and could create a possible point of confusion. I get that it's good practice to be explicit, this is supposed to be a very basic example, so I think we should just keep it very basic.